src/prototypecomponentsnavigatorwindow.cpp

```diff
@@ -0,0 +1,39 @@
+#include "prototypecomponentsnavigatorwindow.h"
+#include "ui_prototypecomponentsnavigatorwindow.h"
+#include <QStandardItemModel>
+#include <QItemSelectionModel>
+
+PrototypeComponentsNavigatorWindow::PrototypeComponentsNavigatorWindow(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::PrototypeComponentsNavigatorWindow)
+{
+    ui->setupUi(this);
+
+    componentPreview = new QStandardItemModel(this);
```

Comment:
This model should not be here. You have setModel() method in your interface - use it for setting model. See architetural notes in your ticket in comments.